Makefile

```diff
@@ -361,82 +279,133 @@ generate-go-conversions-kcp: $(CONVERSION_GEN)
+		find . -name "go.mod" -exec grep -i 'k8s.io/client-go.*+incompatible' {} \; -print; \
+		echo "go module contains an incompatible client-go version"; exit 1; \
+	fi
+
+.PHONY: verify-gen
+verify-gen: generate  ## Verfiy go generated files are up to date
+	@if !(git diff --quiet HEAD); then \
+		git diff; \
+		echo "generated files are out of date, run make generate"; exit 1; \
+	fi
+
+.PHONY: verify-conversions
+verify-conversions: $(CONVERSION_VERIFIER)  ## Verifies API expected conversion are in place
+	$(CONVERSION_VERIFIER)
+
+.PHONY: verify-boilerplate
+verify-boilerplate: ## Verify boilerplate text exists in each file
+	./hack/verify-boilerplate.sh
+
+.PHONY: verify-shellcheck
+verify-shellcheck: ## Verify shell files
+	./hack/verify-shellcheck.sh
+
+.PHONY: verify-tiltfile
+verify-tiltfile: ## Verify Tiltfile format
+	./hack/verify-starlark.sh
 
-## --------------------------------------
-## Modules
-## --------------------------------------
+.PHONY: verify-docker-provider
+verify-docker-provider:
+	@echo "Verifying CAPD"
+	cd $(CAPD_DIR); $(MAKE) verify
 
-.PHONY: modules
-modules: ## Runs go mod to ensure modules are up to date.
-	go mod tidy
-	cd $(TOOLS_DIR); go mod tidy
-	cd $(TEST_DIR); go mod tidy
+.PHONY: verify-book-links
+verify-book-links:
+	$(MAKE) -C docs/book verify
 
 ## --------------------------------------
-## Docker
+## Binaries
 ## --------------------------------------
 
-.PHONY: docker-pull-prerequisites
-docker-pull-prerequisites:
-	docker pull docker.io/docker/dockerfile:1.1-experimental
-	docker pull $(GO_CONTAINER_IMAGE)
-	docker pull gcr.io/distroless/static:latest
+##@ build:
+
+.PHONY: clusterctl
+clusterctl: ## Build the clusterctl binary
+	go build -trimpath -ldflags "$(LDFLAGS)" -o $(BIN_DIR)/clusterctl sigs.k8s.io/cluster-api/cmd/clusterctl
+
+.PHONY: managers
+managers: ## Run all manager-* targets
+	$(MAKE) manager-core
+	$(MAKE) manager-kubeadm-bootstrap
+	$(MAKE) manager-kubeadm-control-plane
+
+.PHONY: manager-core
+manager-core: ## Build the core manager binary into the ./bin folder
+	go build -trimpath -ldflags "$(LDFLAGS)" -o $(BIN_DIR)/manager sigs.k8s.io/cluster-api
+
+.PHONY: manager-kubeadm-bootstrap
+manager-kubeadm-bootstrap: ## Build the kubeadm bootstrap manager binary into the ./bin folder
+	go build -trimpath -ldflags "$(LDFLAGS)" -o $(BIN_DIR)/kubeadm-bootstrap-manager sigs.k8s.io/cluster-api/bootstrap/kubeadm
+
+.PHONY: manager-kubeadm-control-plane
+manager-kubeadm-control-plane: ## Build the kubeadm control plane manager binary into the ./bin folder
+	go build -trimpath -ldflags "$(LDFLAGS)" -o $(BIN_DIR)/kubeadm-control-plane-manager sigs.k8s.io/cluster-api/controlplane/kubeadm
 
 .PHONY: docker-build
-docker-build: docker-pull-prerequisites ## Build the docker images for controller managers
+docker-build: docker-pull-prerequisites ## Run docker-build-core and all docker-build-kubeadm-* targets
```

Comment:
What about adding ALL_DOCKER_BUILD variable and use $(addprefix docker-build-,$(ALL_DOCKER_BUILD))

Reply:
+1 Added